include/cpr/threadpool.h

```diff
@@ -0,0 +1,122 @@
+#ifndef CPR_THREAD_POOL_H
+#define CPR_THREAD_POOL_H
+
+#include <atomic>
+#include <chrono>
+#include <condition_variable>
+#include <functional>
+#include <future>
+#include <list>
+#include <memory>
+#include <mutex>
+#include <queue>
+#include <thread>
+#include <utility>
+
+#define CPR_DEFAULT_THREAD_POOL_MIN_THREAD_NUM  1
+#define CPR_DEFAULT_THREAD_POOL_MAX_THREAD_NUM  std::thread::hardware_concurrency()
+#define CPR_DEFAULT_THREAD_POOL_MAX_IDLE_TIME   60000 // ms
+
+namespace cpr {
+
+class ThreadPool {
+public:
+    using Task = std::function<void()>;
+
+    ThreadPool( int min_threads = CPR_DEFAULT_THREAD_POOL_MIN_THREAD_NUM,
```

Comment:
Since it does not make sense to have a negative amount of threads, please use `size_t` here instead of int.